### PR TITLE
Make focus state clearer on some controls

### PR DESCRIPTION
**Pull request by taipeicoder**
#### Changes proposed in this Pull Request

* Updated CSS on `AppsBadge` and `Task` ("Hide this" button) components so that their focus state is clearer. 

#### Testing instructions

"Hide this" button:
* Head to My Home.
* Click on a card and press tab until the "Hide this" button has focus.
* When the "Hide this" button receives focus, it should have a blue solid outline.
Screen Shot 2022-02-15 at 5 03 36 PM

`AppsBadge`:
* Head to My Home.
* On the right sidebar, beneath Quick Links, find the section Wordpress apps.
* Click on the title "Wordpress apps" and press tab until the `AppsBadge` component receives focus.
* When the `AppsBadge` component receives focus, it should have the browser's default outline (dotted black outline).
Screen Shot 2022-02-15 at 5 17 56 PM

Related to #53617

### CODE REVIEW

**Comment by taipeicoder** [comment]
Two things I would like to point out:

1. There is an existing issue where pressing tabs on a card in the primary `DotPager` after its last focusable element will cause the page to unintentionally move from its position. We can track this in a new GitHub issue.
2. There seems to be a custom outline style which is the blue solid outline that is applied to `input` and `button` elements, should `AppBadges` also have this custom outline style?

File: client/blocks/get-apps/apps-badge.scss [hunk 1]
```diff
@@ -2,9 +2,17 @@
...
 	max-width: 135px;
 	max-height: 40px;
-	overflow: hidden;
 	margin-left: 0.5rem;
 
+	// Set overflow: hidden here instead of the parent node so that its focus outline is not cut-off.
+	> a {
+		overflow: hidden;
+
+		&:focus {
```
Comment (arthur791004) [inline client/blocks/get-apps/apps-badge.scss]: I think using `:focus-visible` is better for this a11y issue if the browser supports

**Review by arthur791004 — COMMENTED** [review]
It might be better if we can also make focus state on the following elements

image

image

image

File: client/blocks/get-apps/apps-badge.scss (hunk 1 above)
Comment (taipeicoder) [reply client/blocks/get-apps/apps-badge.scss]: Added the suggested focus states :-)

File: client/my-sites/customer-home/style.scss [hunk 2]
```diff
@@ -238,6 +238,17 @@ body.is-section-home.theme-default.color-scheme {
...
 }
 
+.primary__customer-home-location-content .dot-pager__control-choose-page,
+.primary__customer-home-location-content .dot-pager__control-prev,
+.primary__customer-home-location-content .dot-pager__control-next,
+.customer-home__card .dot-pager__control-choose-page,
+.customer-home__card .dot-pager__control-prev,
+.customer-home__card .dot-pager__control-next {
+	&:focus-visible {
+		box-shadow: 0 0 0 2px var( --color-primary );
```
Comment (arthur791004) [inline client/my-sites/customer-home/style.scss:248]: I think we can do this in `dot-pager` component itself. See https://github.com/Automattic/wp-calypso/blob/0ecb6b29ef6241aa938d62ec441e0e7794f5d59f/client/components/dot-pager/style.scss#L41

Comment (taipeicoder) [reply client/my-sites/customer-home/style.scss:248]: That crossed my mind, but I'm not 100% sure that the customized outline is something unique to My Home or other pages as well, which is why I only added the override in the My Home page CSS. Does that make sense?

File: client/my-sites/customer-home/cards/tasks/site-setup-list/style.scss [hunk 3]
```diff
@@ -193,6 +193,12 @@
 		margin-top: 0;
 	}
 
+	.site-setup-list__task-skip {
+		&:focus {
+			box-shadow: 0 0 0 2px var( --color-primary );
```
Comment (arthur791004) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/style.scss:199]: Prefer `focus-visible` but it's not blocking

File: client/my-sites/customer-home/cards/tasks/style.scss [hunk 4]
```diff
@@ -87,10 +87,18 @@
 		padding-top: 0;
 		margin-top: auto;
 		font-size: $font-body-small;
+
+		.is-link {
+			padding: 0 8px;
+
+			&:focus {
+				box-shadow: 0 0 0 2px var( --color-primary );
```
Comment (arthur791004) [inline client/my-sites/customer-home/cards/tasks/style.scss:95]: The same as previous one

File: client/my-sites/customer-home/style.scss (hunk 2 above)
Comment (arthur791004) [reply client/my-sites/customer-home/style.scss:248]: I think our [_reset.scss](https://github.com/Automattic/wp-calypso/blob/0ecb6b29ef6241aa938d62ec441e0e7794f5d59f/client/assets/stylesheets/shared/_reset.scss#L128) removed the outline. So it's not customized outline and it's good to apply to component itself.

**Comment by taipeicoder** [comment]
Creating a new PR due unnecessary rebase. Apologies!

**Pull request closed without merging**